arangod/RocksDBEngine/RocksDBMethods.h

```diff
@@ -232,6 +232,33 @@ class RocksDBBatchedMethods final : public RocksDBMethods {
   rocksdb::WriteBatchWithIndex* _wb;
 };
 
+// INDEXING MAY ONLY BE DISABLED IN TOPLEVEL AQL TRANSACTIONS
+// THIS IS BECAUSE THESE TRANSACTIONS WILL EITHER READ FROM
+// OR (XOR) WRITE TO A COLLECTION. IF THIS PRECONDITION IS
+// VIOLATED THE DISABLED INDEXING WILL BREAK GET OPERATIONS.
+struct IndexingDisabler {
+  // will only be active if condition is true
+
+  IndexingDisabler() = delete;
+  IndexingDisabler(IndexingDisabler&&) = delete;
+
+  IndexingDisabler(RocksDBMethods* meth, bool condition = true) : _meth(nullptr) {
```

Comment:
the condition should not be an optional parameter. it should be a mandatory parameter.